### PR TITLE
add support for alternate rpmdb pathes

### PR DESCRIPTION
**Pull request by xaki23**
partial fix for https://github.com/QubesOS/qubes-issues/issues/7346

### CODE REVIEW

File: package-managers/qubes-download-dom0-updates.sh [hunk 1]
```diff
@@ -72,15 +72,15 @@ fi
...
-    # Default rpm configuration on Debian uses ~/.rpmdb for rpm database (as
-    # rpm isn't native package manager there)
-    mkdir -p "$DOM0_UPDATES_DIR$HOME"
-    rm -rf "$DOM0_UPDATES_DIR$HOME/.rpmdb"
-    cp -r "$DOM0_UPDATES_DIR/var/lib/rpm" "$DOM0_UPDATES_DIR$HOME/.rpmdb"
+# Check if we need to copy rpmdb somewhere else
+DBPATH=`rpm -E '%{_dbpath}'`
+if [ ! "$DBPATH" = "/var/lib/rpm" ]; then
+    mkdir -p "$DOM0_UPDATES_DIR$DBPATH"
+    rm -rf "$DOM0_UPDATES_DIR$DBPATH"
```
Comment (DemiMarie) [inline package-managers/qubes-download-dom0-updates.sh:79]: I suggest checking that `$DOM0_UPDATES_DIR` is something reasonable before running the `rm -rf`.

Comment (xaki23) [reply package-managers/qubes-download-dom0-updates.sh:79]: note the rm -rf isnt a new part there, i just made the so-far debian-specific kludge more generic.
also, this is run as user, so the damage potential is ... limited.

Comment (DemiMarie) [reply package-managers/qubes-download-dom0-updates.sh:79]: Good point!  Withdrawn.

**Review by DemiMarie — APPROVED** [review]

File: package-managers/qubes-download-dom0-updates.sh [hunk 2]
```diff
@@ -72,15 +72,15 @@ fi
...
 mkdir -p $DOM0_UPDATES_DIR/etc
 
-if [ -e /etc/debian_version ]; then
-    # Default rpm configuration on Debian uses ~/.rpmdb for rpm database (as
-    # rpm isn't native package manager there)
-    mkdir -p "$DOM0_UPDATES_DIR$HOME"
-    rm -rf "$DOM0_UPDATES_DIR$HOME/.rpmdb"
-    cp -r "$DOM0_UPDATES_DIR/var/lib/rpm" "$DOM0_UPDATES_DIR$HOME/.rpmdb"
+# Check if we need to copy rpmdb somewhere else
+DBPATH=`rpm -E '%{_dbpath}'`
```
Comment (DemiMarie) [inline package-managers/qubes-download-dom0-updates.sh:76]:
```suggestion
DBPATH=$(rpm --eval '%{_dbpath}')
```

shellcheck complains about the use of backticks, and `--eval` is more readable than `-E`.

File: package-managers/qubes-download-dom0-updates.sh [hunk 3]
```diff
@@ -72,15 +72,15 @@ fi
...
+if [ ! "$DBPATH" = "/var/lib/rpm" ]; then
+    mkdir -p "$DOM0_UPDATES_DIR$DBPATH"
+    rm -rf "$DOM0_UPDATES_DIR$DBPATH"
+    cp -r "$DOM0_UPDATES_DIR/var/lib/rpm" "$DOM0_UPDATES_DIR$DBPATH"
 fi
 # Rebuild rpm database in case of different rpm version
-rm -f $DOM0_UPDATES_DIR/var/lib/rpm/__*
+rm -f $DOM0_UPDATES_DIR$DBPATH/__*
```
Comment (DemiMarie) [inline package-managers/qubes-download-dom0-updates.sh:83]:
```suggestion
rm -f -- "$DOM0_UPDATES_DIR$DBPATH"/__*
```

shellcheck complains

**Review by DemiMarie — CHANGES_REQUESTED** [review]
Shellcheck complains, causing CI to fail.

File: package-managers/qubes-download-dom0-updates.sh (hunk 3 above)
Comment (xaki23) [reply package-managers/qubes-download-dom0-updates.sh:83]: why does it want to add -- for one rm, but not the other one? :thinking:

Comment (DemiMarie) [reply package-managers/qubes-download-dom0-updates.sh]: That was my suggestion, not ShellCheck’s.  It isn’t strictly necessary.

Comment (xaki23) [reply package-managers/qubes-download-dom0-updates.sh]: ah! added it to both rm calls while i was at it.

**Review by DemiMarie — APPROVED** [review]

**Comment by DemiMarie** [comment]
PipelineRetry

**Review by marmarek — APPROVED** [review]

**Pull request merged**